exchanges/kraken/kraken_wrapper.go

```diff
@@ -328,60 +328,105 @@ func (k *Kraken) Run(ctx context.Context) {
 		}
 	}
 
-	if !k.GetEnabledFeatures().AutoPairUpdates && !forceUpdate {
-		return
+	if k.GetEnabledFeatures().AutoPairUpdates || forceUpdate {
+		if err := k.UpdateTradablePairs(ctx, forceUpdate); err != nil {
+			log.Errorf(log.ExchangeSys,
+				"%s failed to update tradable pairs. Err: %s",
+				k.Name,
+				err)
+		}
+	}
+
+	for _, a := range k.GetAssetTypes(true) {
+		if err := k.UpdateOrderExecutionLimits(ctx, a); err != nil && err != common.ErrNotYetImplemented {
+			log.Errorln(log.ExchangeSys, err.Error())
+		}
+	}
+}
+
+// UpdateOrderExecutionLimits sets exchange execution order limits for an asset type
+func (k *Kraken) UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) error {
+	if a != asset.Spot {
+		return common.ErrNotYetImplemented
+	}
+
+	pairInfo, err := k.fetchSpotPairInfo(ctx)
+	if err != nil {
+		return fmt.Errorf("%s failed to load %s pair execution limits. Err: %s", k.Name, a, err)
+	}
+
+	limits := make([]order.MinMaxLevel, 0, len(pairInfo))
+
+	for pair, info := range pairInfo {
+		limits = append(limits, order.MinMaxLevel{
+			Asset:                  a,
+			Pair:                   pair,
+			PriceStepIncrementSize: info.TickSize,
+			MinimumBaseAmount:      info.OrderMinimum,
+		})
 	}
 
-	err := k.UpdateTradablePairs(ctx, forceUpdate)
+	if err := k.LoadLimits(limits); err != nil {
+		return fmt.Errorf("%s Error loading %s exchange limits: %v", k.Name, a, err)
```

Comment:
If you use `%w` instead of `%v` of `%s` for errors when using `fmt.Errorf`, you get to wrap the original error type instead of declaring a new one. While Its not important in this instance, it can allow for easier debugging to identify the type of problem, especially if the error text is lacking

For instance, if `limits` contained an unsupported asset, you can still check if the returned error was asset.ErrNotSupported, but when using `%s` you cannot without string inspection

Reply:
that is completed